### DISCUSSION TITLE
new web interface

### DISCUSSION THREAD

**Discussion by ghost** [mcguirepr89/BirdNET-Pi#44] [Ideas]
Thousands of easy HTML5 webpages are free available which can easily be adopted to the simple needs. The simplest form could be a page with a top menu where all the current entries can be selected and a main page below. An internet recherche for an ideal template should first be done, followed with an adoption for the links.

**Comment by ghost** [comment]
From the technical point of view the old HTML **frameset** element has to be used. This is an example: --> https://www.quackit.com/html/templates/frames/ (bottom).


![Bildschirmfoto 2021-10-18 um 04 41 16](https://user-images.githubusercontent.com/78355888/137661479-b14ec99b-d8e7-4f11-a577-8490006a1295.png)

The adaption has to be finished and the links have to be set. Perhaps images and a "cooperate identity design" has to be included.

![Bildschirmfoto 2021-10-18 um 04 48 38](https://user-images.githubusercontent.com/78355888/137661790-21b5d805-a7b7-4985-8983-2a259df59eae.png)

I will continue on that tomorrow...

**Comment by ghost** [comment]


**Reply by mcguirepr89** [reply]
Wow! That was a lot of progress very quickly. I will try to catch up to you today and see what I can mock up. Please upload that most recent index.html file if you are able -- that would be a great place for me to start

**Reply by ghost** [reply]
Attached you can find the html sources including one image. I consist only of a few small html pages. All can be exchanged (text, colors, image...). Let me know what you think and what can be done better. I have not tested it on mobile devices yet.

[html.zip](https://github.com/mcguirepr89/BirdNET-Pi/files/7373071/html.zip)


![Bildschirmfoto 2021-10-19 um 10 55 35](https://user-images.githubusercontent.com/78355888/137877322-fb5a217c-d42d-4a07-aa68-f373da3e57bc.png)

**Reply by ghost** [reply]
TBD: in the menu.html the distance between the `heading` and the `href` elements has to be decreased in the css `style` section.

**Reply by ghost** [reply]
latest version:

[html.zip](https://github.com/mcguirepr89/BirdNET-Pi/files/7375220/html.zip)

![Bildschirmfoto 2021-10-19 um 18 38 19](https://user-images.githubusercontent.com/78355888/137954406-fb7a9dc5-57e2-4f99-aae5-cb4039285e33.png)

**Comment by mcguirepr89** [comment]
![Screenshot from 2021-10-20 16-45-44](https://user-images.githubusercontent.com/60325264/138169565-116eda84-540e-497b-9c79-c5e15c58a23b.png)
Getting it functioning now and tweaking some things -- I expect to make this an update for testers probably tomorrow.

Thanks for all your hard work on this -- I am really liking the way it is turning out :)

**Comment by ghost** [comment]
👍👍👍

**Reply by mcguirepr89** [reply]
I love the way it looks on the desktop, but it is awful on my phone. I think I'll be retaining the old index.html as the "Mobile View" ;)

**Reply by ghost** [reply]
you can distinguish between desktop and mobile and then load different pages.

**Reply by mcguirepr89** [reply]
yep that's what i've configured -- works like a charm! :sunglasses:

**Comment by mcguirepr89** [comment]
![Screenshot from 2021-10-22 16-53-57](https://user-images.githubusercontent.com/60325264/138521770-8801d5e4-f1e6-42e1-852e-244bdfa13ac7.png)

**Reply by ghost** [reply]
better!

**Reply by mcguirepr89** [reply]
And the final change is to those links -- 
![Screenshot from 2021-10-22 17-17-39](https://user-images.githubusercontent.com/60325264/138523995-5eb45173-a7fd-4986-b929-b1a760ecf4cc.png)

**Reply by ghost** [reply]
The last kick is that the bird could be an animated gif.

**Reply by mcguirepr89** [reply]
OR clicking the bird plays the live stream???????

**Reply by ghost** [reply]
yeah, thats perfect.

**Reply by mcguirepr89** [reply]
The GIF should be for the waiting pages for backups and updates! Class :sunglasses:

**Reply by mcguirepr89** [reply]
have you tested the backup feature yet? It will have a partner feature which will restore from those backups

**Reply by ghost** [reply]
not tested yet

**Reply by mcguirepr89** [reply]
Any chance you would be able to make another GIF of the system for the README.md sometime? I don't think there will be TOO many more cosmetic changes or name changing or anything like that anytime soon. Although this discussion only opened 5 days ago and a lot has changed since then, so it's hard to predict the future.

**Reply by ghost** [reply]
sure, i'm just waiting for the final outfit.

**Reply by mcguirepr89** [reply]
good thinking
![Screenshot from 2021-10-22 17-58-18](https://user-images.githubusercontent.com/60325264/138527473-18092831-63fd-4875-a765-f255b7b98332.png)

**Reply by mcguirepr89** [reply]
no more scroll bar (it's hidden) :)
